tests/tpubkey

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Jakub Jelen <jjelen@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source ${TESTSSRCDIR}/helpers.sh
+
+TSTOUT="${TMPPDIR}/testout"
+
+title PARA "Export RSA Public key to a file"
+ossl 'pkey -in $BASEURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export Public key to a file (pub-uri)"
+ossl 'pkey -in $PUBURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export Public key to a file (with pin)"
+ossl 'pkey -in $BASEURIWITHPIN -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Print Public key from private"
+ossl 'pkey -in $PRIURI -pubout -text' output
+FAIL=0
+echo $output | grep "PKCS11 RSA Public Key (2048 bits)" > /dev/null 2>&1 || FAIL=1
+if [ $FAIL -eq 1 ]; then
+    echo "Pkcs11 encoder function failed"
+    echo
+    echo "Original command output:"
+    echo "$output"
+    echo
+    exit 1
+fi
+
+title PARA "Export Public check error"
+FAIL=0
+ossl 'pkey -in pkcs11:id=%de%ad -pubin
+           -pubout -out ${TSTOUT}-invlid.pub' || FAIL=1
+if [ $FAIL -eq 0 ]; then
+    echo "Invalid pkcs11 uri resulted in no error exporting key"
+    exit 1
+fi
+
+title PARA "Export EC Public key to a file"
+ossl 'pkey -in $ECBASEURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export EC Public key to a file (pub-uri)"
+ossl 'pkey -in $ECPUBURI -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Export EC Public key to a file (with pin)"
+ossl 'pkey -in $ECBASEURIWITHPIN -pubin -pubout -out ${TSTOUT}.pub'
+title LINE "Print EC Public key from private"
+ossl 'pkey -in $ECPRIURI -pubout -text' output
+FAIL=0
+echo $output | grep "PKCS11 EC Public Key (256 bits)" > /dev/null 2>&1 || FAIL=1
+if [ $FAIL -eq 1 ]; then
+    echo "Pkcs11 encoder function failed"
+    echo
+    echo "Original command output:"
+    echo "$output"
+    echo
+    exit 1
+fi
+
+# The softokn does not support removing only public key from the key pair
+if [ ! -z "$BASE2URI" ]; then
+    title PARA "Check we can get public keys from certificate objects"
+
+    title LINE "Export Public key to a file (priv-uri)"
+    ossl 'pkey -in $PRI2URI -pubin -pubout -out ${TSTOUT}-priv-cert.pub'
```

Comment:
remove -pubin from this one

Reply:
Unfortunately this does not use the certificate fallback, because RSA Private keys include data for the public key, so there is no fallback.

It may work right using EC keys as private EC key PKCS#11 objects generally do not include public key material in their attributes

Reply:
Right, the privuri is probably not useful as from the EC private key we can not get the public key. But again, we could get it from the certificate, if it exists and serve it to the calling application.

Reply:
It is exaclty this that makes EC useful, you pass a private key, then try to extract the public key, the fallback code will try to find an associated public key, fail, then try an associated certificate, succeed, and try to extract the public key from the certificate.

Reply:
So let me extend this to the EC.